File: frontend/src/config/paths.ts
/**
 * Type-safe path builders for navigation
 * Single source for constructing URLs throughout the app
 */
export const paths = {
  home: () => '/', // mobile OK
  login: () => '/login', // mobile OK
  deviceVerifyGarmin: () => '/garmin',
  deviceVerifyKaroo: () => '/karoo',
  verifyEmail: () => '/verify-email', // mobile OK
  forgotPassword: () => '/forgot-password', // mobile OK
  resetPassword: () => '/reset-password', // mobile OK
  privacy: () => '/privacy', // mobile OK
  terms: () => '/terms', // mobile OK
  profile: () => '/profile', // mobile OK
  calendar: () => '/calendar', // mobile OK
  allRoutes: () => '/routes',

  // Teams
  teams: () => '/teams', // mobile OK
  teamsNew: () => '/teams/new',
  team: (teamSlug: string) => `/teams/${teamSlug}`, // mobile OK
  teamAbout: (teamSlug: string) => `/teams/${teamSlug}/about`, // mobile OK (redirects to team)
  teamCalendar: (teamSlug: string) => `/teams/${teamSlug}/calendar`, // mobile OK

  // Team Pages (displayed as tabs)
  teamPage: (teamSlug: string, pageSlug: string) => `/teams/${teamSlug}/pages/${pageSlug}`,

  // Team Admin
  teamAdmin: (teamSlug: string) => `/teams/${teamSlug}/admin`,
  teamAdminPlaces: (teamSlug: string) => `/teams/${teamSlug}/admin/places`,
  teamAdminPages: (teamSlug: string) => `/teams/${teamSlug}/admin/pages`,
  teamAdminPageNew: (teamSlug: string) => `/teams/${teamSlug}/admin/pages/new`,
  teamAdminPageEdit: (teamSlug: string, pageSlug: string) =>
    `/teams/${teamSlug}/admin/pages/${pageSlug}/edit`,
  teamMembers: (teamSlug: string) => `/teams/${teamSlug}/admin/members`,
  teamSettings: (teamSlug: string) => `/teams/${teamSlug}/admin/settings`,

  // Rides
  rideNew: (teamSlug: string) => `/teams/${teamSlug}/rides/new`,
  ride: (teamSlug: string, rideSlug: string) => `/teams/${teamSlug}/rides/${rideSlug}`, // mobile OK
  rideEdit: (teamSlug: string, rideSlug: string) => `/teams/${teamSlug}/rides/${rideSlug}/edit`,

  // Ride Templates
  rideTemplates: (teamSlug: string) => `/teams/${teamSlug}/admin/ride-templates`,
  rideTemplateNew: (teamSlug: string) => `/teams/${teamSlug}/admin/ride-templates/new`,
  rideTemplateEdit: (teamSlug: string, templateSlug: string) =>
    `/teams/${teamSlug}/admin/ride-templates/${templateSlug}/edit`,

  // Trips
  tripNew: (teamSlug: string) => `/teams/${teamSlug}/trips/new`,
  trip: (teamSlug: string, tripSlug: string) => `/teams/${teamSlug}/trips/${tripSlug}`,
  tripEdit: (teamSlug: string, tripSlug: string) => `/teams/${teamSlug}/trips/${tripSlug}/edit`,
  stage: (teamSlug: string, tripSlug: string, stageSlug: string) =>
    `/teams/${teamSlug}/trips/${tripSlug}/stages/${stageSlug}`,

  // Posts
  postNew: (teamSlug: string) => `/teams/${teamSlug}/posts/new`,
  post: (teamSlug: string, postSlug: string) => `/teams/${teamSlug}/posts/${postSlug}`,
  postEdit: (teamSlug: string, postSlug: string) => `/teams/${teamSlug}/posts/${postSlug}/edit`,

  // Routes
  routes: (teamSlug: string) => `/teams/${teamSlug}/routes`, // mobile OK
  routeNew: (teamSlug: string) => `/teams/${teamSlug}/routes/new`,
  route: (teamSlug: string, routeSlug: string) => `/teams/${teamSlug}/routes/${routeSlug}`, // mobile OK
  routeEdit: (teamSlug: string, routeSlug: string) => `/teams/${teamSlug}/routes/${routeSlug}/edit`,

  // Ads
  ads: (teamSlug: string) => `/teams/${teamSlug}/ads`,
  adNew: (teamSlug: string) => `/teams/${teamSlug}/ads/new`,
  ad: (teamSlug: string, adSlug: string) => `/teams/${teamSlug}/ads/${adSlug}`,
  adEdit: (teamSlug: string, adSlug: string) => `/teams/${teamSlug}/ads/${adSlug}/edit`,

  // Platform Admin
  admin: () => '/admin',
  adminDomains: () => '/admin/domains',
  adminTeams: () => '/admin/teams',
  adminUsers: () => '/admin/users',
} as const

File: mobile/lib/config/paths.dart
/// Type-safe path builders for navigation
/// Single source for constructing URLs throughout the app
/// Mirrors frontend/src/config/paths.ts
class Paths {
  Paths._();

  // Main tabs
  static String home() => '/home'; // applink OK (/)
  static String teams() => '/teams'; // applink OK
  static String calendar() => '/calendar'; // applink OK
  static String profile() => '/profile'; // applink OK

  // Legal
  static String privacy() => '/privacy'; // applink OK
  static String terms() => '/terms'; // applink OK

  // Auth
  static String login() => '/login'; // applink OK
  static String register() => '/register'; // applink OK
  static String verifyEmail() => '/verify-email'; // applink OK
  static String forgotPassword() => '/forgot-password'; // applink OK
  static String resetPassword() => '/reset-password'; // applink OK

  // Device verification
  static String deviceVerifyGarmin() => '/garmin'; // applink OK
  static String deviceVerifyKaroo() => '/karoo'; // applink OK

  // Teams
  static String team(String teamSlug) => '/teams/$teamSlug'; // applink OK
  static String teamFeed(String teamSlug) => '/teams/$teamSlug/feed';
  static String teamCalendar(String teamSlug) => '/teams/$teamSlug/calendar'; // applink OK
  static String teamAbout(String teamSlug) => '/teams/$teamSlug/about'; // applink OK
  static String teamAds(String teamSlug) => '/teams/$teamSlug/ads';

  // Rides
  static String ride(String teamSlug, String rideSlug) =>
      '/teams/$teamSlug/rides/$rideSlug'; // applink OK

  // Posts
  static String post(String teamSlug, String postSlug) =>
      '/teams/$teamSlug/posts/$postSlug'; // applink OK

  // Routes
  static String routes(String teamSlug) => '/teams/$teamSlug/routes'; // applink OK
  static String route(String teamSlug, String routeSlug) =>
      '/teams/$teamSlug/routes/$routeSlug'; // applink OK

  // Ads
  static String ad(String teamSlug, String adSlug) =>
      '/teams/$teamSlug/ads/$adSlug'; // applink OK

  // Trips
  static String trip(String teamSlug, String tripSlug) =>
      '/teams/$teamSlug/trips/$tripSlug'; // applink OK
  static String stage(String teamSlug, String tripSlug, String stageSlug) =>
      '/teams/$teamSlug/trips/$tripSlug/stages/$stageSlug'; // applink OK
}
